Sources/FoundationEssentials/Data/Data+Base64.swift

```diff
@@ -0,0 +1,374 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+private enum Base64Error: Error {
+    case invalidElementCount
+    case cannotDecode
+}
+
+@available(macOS 10.10, iOS 8.0, watchOS 2.0, tvOS 9.0, *)
+extension Data {
+
+    // MARK: - Init from base64
+
+    /// Initialize a `Data` from a Base-64 encoded String using the given options.
+    ///
+    /// Returns nil when the input is not recognized as valid Base-64.
+    /// - parameter base64String: The string to parse.
+    /// - parameter options: Encoding options. Default value is `[]`.
+    public init?(
+        base64Encoded base64String: consuming String,
+        options: Base64DecodingOptions = []
+    ) {
+        do {
+            self = try base64String.withUTF8 {
+                // String won't pass an empty buffer with a `nil` `baseAddress`.
+                try Data(
+                    decodingBase64: BufferView(unsafeBufferPointer: $0)!,
+                    options: options
+                )
+            }
+        }
+        catch {
+            precondition(error is Base64Error)
+            return nil
+        }
+    }
+
+
+    /// Initialize a `Data` from a Base-64, UTF-8 encoded `Data`.
+    ///
+    /// Returns nil when the input is not recognized as valid Base-64.
+    ///
+    /// - parameter base64Data: Base-64, UTF-8 encoded input data.
+    /// - parameter options: Decoding options. Default value is `[]`.
+    public init?(
+        base64Encoded base64Data: borrowing Data,
```

Comment:
Is this ABI compatible with the existing public API?

```
    @inlinable // This is @inlinable as a convenience initializer.
    public init?(base64Encoded base64Data: __shared Data, options: Data.Base64DecodingOptions = [])
```

Reply:
`__shared` and `borrowing` are ABI-compatible. `__owned` and `consuming` are not.

Reply:
I reverted to `__shared` after checking on the actual mangling.